### PR TITLE
Add player count to MP lobby, prevent too many players

### LINKED ISSUE #506: Multiplayer handling content set mismatch & too many players
1. transmit content sets as part of initial multiplayer message
2. have a dialog for warning folks when there's a content set mismatch, w/ information on the lowest common denominator and how to change (+ block connecting)
3. once you're in a multiplayer session, disable the change content sets button

This way, we don't have to continuously transmit content sets or re-calculate local vs all content sets - once you're in a session, you know you're all locked to the same content set.

Similarly, for number of players, we should show a dialog that says too many players in the session, with a player count selector so that they can change + continue to join; once in a multiplayer session, all player count selectors (aka that selector, plus the game mode screen + settings screen) should be mindful of the total number of multiplayer players, preventing you from increasing it over 6 + showing a note in the subtext when the mp player count is 6 / session is already full.

### PR DESCRIPTION
**Pull request by smartin015**
Over size:
image

Correct size:
image

Fixes #506 

Includes tests for `<Picker/>` and `ctx._.numAdventurers()`.

### CODE REVIEW

File: services/app/src/components/base/PlayerCount.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    if (delta < 0 && this.props.localPlayers + delta < 1) {
+      return;
+    }
+    this.props.onChange(this.props.localPlayers + delta);
+  }
+
+  public render() {
+    return (
+      <Picker id={this.props.id} label="Players" value={this.props.localPlayers} onDelta={(i: number) => this.onDelta(i)}>
+        {(this.props.allPlayers > 1) ? 'The number of players.' : <div><strong>Solo play:</strong> Play as two adventurers with double the combat timer.</div>}
```
Comment (toddmedema) [inline services/app/src/components/base/PlayerCount.tsx]: Technically not just the combat timer, affects the decision timer too... Not quite sure how to say that concisely tho

File: services/app/src/components/views/Multiplayer.tsx [hunk 2]
```diff
@@ -108,8 +113,12 @@ export function renderLobby(props: Props): JSX.Element {
...
         </table>
         <p>Click the connected adventurers or connection state for more information.</p>
-        <p>Once everyone is connected, click Start:</p>
-        <Button id="start" className="mediumbutton" onClick={() => {props.onStart(); }}>Start</Button>
+        <h2>Adventurers</h2>
+        <p>Max of {MAX_ADVENTURERS} players allowed in multiplayer.</p>
+        <PlayerCount id="playerCount" localPlayers={props.settings.numLocalPlayers} allPlayers={allPlayers} onChange={(localPlayers: number) => props.onPlayerChange(localPlayers)} />
+        <br/>
+        <p>Once everyone is ready, click Start:</p>
+        <Button id="start" className="mediumbutton" disabled={allPlayers > MAX_ADVENTURERS} onClick={() => {props.onStart(); }}>{(allPlayers > MAX_ADVENTURERS) ? `Player count must be < ${MAX_ADVENTURERS}` : 'Start'}</Button>
```
Comment (toddmedema) [inline services/app/src/components/views/Multiplayer.tsx]: Must be <=

File: services/app/src/components/base/PlayerCount.tsx (hunk 1 above)
Comment (smartin015) [reply services/app/src/components/base/PlayerCount.tsx]: Removed "combat".

File: services/app/src/components/views/Multiplayer.tsx (hunk 2 above)
Comment (smartin015) [reply services/app/src/components/views/Multiplayer.tsx]: Done.

**Pull request merged**